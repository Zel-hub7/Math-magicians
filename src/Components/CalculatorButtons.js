import React from 'react';



function CalculatorButtons({ onButtonClick }) {
   const hadleClick = (buttonName) => {
    onButtonClick(buttonName);
   }
  return (
    <div className="calculator-grid">
      <button type = 'button' onClick={() => handleClick('AC')}>AC</button>
      <button type = 'button' onClick={() => handleClick('+/-')}>+/-</button>
      <button type = 'button' onClick={() => handleClick('%')}>%</button>
      <button type='button' onClick={() => handleClick('AC')} className="oper">/</button>
      <button type = 'button' onClick={() => handleClick('7')}>7</button>
      <button type = 'button' onClick={() => handleClick('8')}>8</button>
      <button type = 'button' onClick={() => handleClick('9')}>9</button>
      <button className="oper" onClick={() => handleClick('*')}>*</button>
      <button type = 'button' onClick={() => handleClick('4')}>4</button>
      <button type = 'button' onClick={() => handleClick('5')}>5</button>
      <button type = 'button' onClick={() => handleClick('6')}>6</button>
      <button className="oper" onClick={() => handleClick('-')}>-</button>
      <button type = 'button' onClick={() => handleClick('1')}>1</button>
      <button type = 'button' onClick={() => handleClick('2')}>2</button>
      <button type = 'button' onClick={() => handleClick('3')}>3</button>
      <button className="oper" onClick={() => handleClick('+')}>+</button>
      <button className="span-two" onClick={() => handleClick('0')}>0</button>
      <button type = 'button' onClick={() => handleClick('.')}>.</button>
      <button type='button' onClick={() => handleClick('=')} className="oper">=</button>
    </div>
  );
}

export default CalculatorButtons;
