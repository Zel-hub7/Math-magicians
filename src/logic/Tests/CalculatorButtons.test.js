import React from 'react';
import { render } from '@testing-library/react';
import CalculatorButtons from '../../Components/CalculatorButtons';

test('CalculatorButtons snapshot', () => {
  const { asFragment } = render(<CalculatorButtons onButtonClick={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
