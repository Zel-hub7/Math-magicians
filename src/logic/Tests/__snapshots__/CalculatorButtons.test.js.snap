// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButtons snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculator-grid"
  >
    <button
      type="button"
    >
      AC
    </button>
    <button
      type="button"
    >
      +/-
    </button>
    <button
      type="button"
    >
      %
    </button>
    <button
      class="oper"
      type="button"
    >
      รท
    </button>
    <button
      type="button"
    >
      7
    </button>
    <button
      type="button"
    >
      8
    </button>
    <button
      type="button"
    >
      9
    </button>
    <button
      class="oper"
    >
      x
    </button>
    <button
      type="button"
    >
      4
    </button>
    <button
      type="button"
    >
      5
    </button>
    <button
      type="button"
    >
      6
    </button>
    <button
      class="oper"
    >
      -
    </button>
    <button
      type="button"
    >
      1
    </button>
    <button
      type="button"
    >
      2
    </button>
    <button
      type="button"
    >
      3
    </button>
    <button
      class="oper"
    >
      +
    </button>
    <button
      class="span-two"
    >
      0
    </button>
    <button
      type="button"
    >
      .
    </button>
    <button
      class="oper"
      type="button"
    >
      =
    </button>
  </div>
</DocumentFragment>
`;
